department-of-veterans-affairs/caseflow
# HunJerBAH/APPEALS-17522: Create IssuesUpdate Task Model

Resolves [APPEALS-17522](https://vajira.max.gov/browse/APPEALS-17522)

### Description
Created task model to capture changes to an issue's MST and/or PACT status.
### Acceptance Criteria
- [ ] The IssuesUpdateTask formats the Task instructions properly to render on the frontend.
- [ ] The IssuesUpdateTask renders the instructions properly when showing on the Case Timeline

### Testing Plan
1. Open up the Rails console by running `cd ~/appeals/caseflow` and `rails console`.
2. "Sign in" as a BVA Intake User by running `RequestStore[:current_user] = User.find_by(css_id: 'BVADWISE')`
3. Find an appeal by running `appeal = Appeal.find(1)`.
4. Create an IssuesUpdateTask by running `iup = IssuesUpdateTask.create!(appeal: appeal, parent: appeal.root_task, assigned_to: Organization.find_by_url("bva-intake"), assigned_by: RequestStore[:current_user])`
5. Once the task is created, create mock instructions by running `iup.format_instructions("test category", false, false, true, false, "reason for change here")`. This will set the **original** special issues status to **None** and the **updated** issues status to **MST**.
6. If you want to mock a payload for PACT, run `iup.format_instructions("test category", false, false, false, true, "reason for change here")` to set the **original** special issues status from **None** to **PACT**.
7. Complete the task by running `iup.completed!`. The task will now be mocked to show up in the Case Timeline of the appeal.

####Testing the Changes in the Front End
1. Sign in as a BVA Intake User and navigate to the Queue app. 
2. Click the **search glass** to search for a veteran's appeal.
3. In the search bar, enter `500000000`. Click the appeal's steam docket number. The Stream Docket Number is `230405-1` on my local. 
4. On the Case Details Page, navigate to the bottom of the page to look at the Case Timeline. Find the IssuesUpdateTask and click the **instructions** link to view the instructions. If you added both updates to the instructions in steps 6 and 7, your screen will look similar to the images below. Confirm one is labeled **MST** and the other is labeled **PACT** under the **Special Issues Update**
![image](https://user-images.githubusercontent.com/99915461/232839992-cb32217b-32d2-4d5f-96ac-66a5e09aa749.png)
![image](https://user-images.githubusercontent.com/99915461/232840060-5e67d8e1-4514-4a72-b475-2e4a5b781ef3.png)
